client/my-sites/domains/email-providers-upsell/index.jsx

```diff
@@ -10,47 +10,53 @@ import { useTranslate } from 'i18n-calypso';
  * Internal dependencies
  */
 import { domainAddNew } from 'calypso/my-sites/domains/paths';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import EmailProvidersComparison from 'calypso/my-sites/email/email-providers-comparison';
 import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
-import HeaderCake from 'calypso/components/header-cake';
-import PromoCard from 'calypso/components/promo-section/promo-card';
-
-/**
- * Style dependencies
- */
-import emailIllustration from 'calypso/assets/images/email-providers/email-illustration.svg';
 
 export default function EmailProvidersUpsell( { domain } ) {
 	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
 	const translate = useTranslate();
 
-	const handleGoBack = () => {
-		page( domainAddNew( selectedSiteSlug ) );
-	};
-
-	const image = {
-		path: emailIllustration,
-		align: 'right',
-	};
+	const comment = '%(domainName)s is the domain name, e.g example.com';
 
 	return (
 		<>
-			<HeaderCake onClick={ handleGoBack }>
-				{ translate( 'Register %(domain)s', { args: { domain } } ) }
-			</HeaderCake>
-
-			<PromoCard
-				isPrimary
-				title={ translate( 'Add a professional email address to %(domainName)s', {
-					args: {
-						domainName: domain,
-					},
-					comment: '%(domainName)s is the domain name, e.g example.com',
-				} ) }
-				image={ image }
-				className="email-providers-upsell__action-panel"
-			>
-				<p>{ translate( 'No setup or software required. Easy to manage from your dashboard.' ) }</p>
-			</PromoCard>
+			<CalypsoShoppingCartProvider>
+				<EmailProvidersComparison
+					backPath={ domainAddNew( selectedSiteSlug ) }
+					cartDomainName={ domain }
+					headerTitle={ ( domainName ) =>
+						translate( 'Register %(domainName)s', { args: { domainName }, comment } )
+					}
```

Comment:
We shouldn't be passing this as a function, but just as the translated string.
```suggestion
					headerTitle={ translate( 'Register %(domainName)s', { args: { domain }, comment } )}
```
We can do that because this code is specifying the domain of interest!
We may also save a bit on (re)renders because the value comparisons for dynamic functions may suggest that we have a new functions for each render cycle.